lib/fast_cache/cache.rb

```diff
@@ -84,6 +85,7 @@ def delete(key)
       entry = @data.delete(key)
       if entry
         @expires_at.delete(entry)
+        @cleanup.call(entry.value) if @cleanup
```

Comment:
Hmm.  As implemented, the value will still be returned here after the cleanup block has been called.  I think that's ok, as long as the user is aware that's the case -- if they are invalidating the object in some way with their cleanup block, they should not treat the returned value here as something usable.  (But, might still be useful for identity comparison or something like that.)

An alternative, I suppose, could be to return `nil` here if there is a cleanup block.